File: app/src/main/java/com/android/debugmvp/MainActivity.java
package com.android.debugmvp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;

import com.android.debugmvp.mvp.MVPActivityImpl;
import com.android.debugmvp.test.BlogLoopActivity;
import com.debug.framework.mvp.IBasePresenter;

public class MainActivity extends MVPActivityImpl {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        startActivity(new Intent(this, BlogLoopActivity.class));
    }

    @Override
    protected IBasePresenter createPresenter() {
        return null;
    }

    @Override
    protected void initView() {

    }
}

File: app/src/main/java/com/android/debugmvp/MainActivity.kt
package com.android.debugmvp

import android.view.View
import com.android.debugmvp.databinding.ActivityMainBinding
import com.android.debugmvp.mvp.MVPActivityImpl
import com.android.debugmvp.test.BlankFragment
import com.android.debugmvp.test.BlogLoopActivity
import com.android.debugmvp.test.PermissionTestActivity
import com.android.debugmvp.test.presenter.EmptyPresenter
import com.android.debugmvp.utils.AppToast
import com.android.lib.common.utils.navActivity
import com.android.lib.common.utils.singleClick
import com.hi.dhl.binding.viewbind

class MainActivity : MVPActivityImpl<EmptyPresenter>() {

    private val binding: ActivityMainBinding by viewbind()

    override fun createPresenter(): EmptyPresenter {
        return EmptyPresenter()
    }

    override fun initView() {
        binding.tvMsg.text = "binding 默认2个例子"

        //进入权限申请页面
        binding.btnPermission.singleClick {
            navActivity<PermissionTestActivity>()
        }
    }

    fun openAct(view: View?) {
        startAct(BlogLoopActivity::class.java)
    }

    fun openFra(view: View?) {
        AppToast.toast("打开fragment")
        val tran = supportFragmentManager.beginTransaction()
        tran.replace(R.id.fl_container, BlankFragment.newInstance("", ""))
        tran.commitNow()
    }

}